core/src/main/java/nl/nn/adapterframework/jdbc/JdbcFacade.java

```diff
@@ -115,43 +117,36 @@ protected DataSource getDatasource() throws JdbcException {
 			}
 
 			String dsinfo = datasource.toString();
-			log.info(getLogPrefix()+"looked up Datasource ["+dsName+"]: ["+dsinfo+"]");
+			log.info("{}looked up Datasource [{}]: [{}]", getLogPrefix(), dsName, dsinfo);
 		}
 		return datasource;
 	}
 
 	public String getDatasourceInfo() throws JdbcException {
-		String dsinfo=null;
 		if(getDatasource() instanceof TransactionalDbmsSupportAwareDataSourceProxy) {
-			// TODO let TransactionalDbmsSupportAwareDataSourceProxy.getInfo() use the same code as used here
 			return ((TransactionalDbmsSupportAwareDataSourceProxy) getDatasource()).getInfo();
 		}
-		try (Connection conn=getConnection()) {
-			DatabaseMetaData md=conn.getMetaData();
-			String product=md.getDatabaseProductName();
-			String productVersion=md.getDatabaseProductVersion();
-			String driver=md.getDriverName();
-			String driverVersion=md.getDriverVersion();
-			String url=md.getURL();
-			String user=md.getUserName();
-			dsinfo ="user ["+user+"] url ["+url+"] product ["+product+"] product version ["+productVersion+"] driver ["+driver+"] driver version ["+driverVersion+"] datasource ["+getDatasource().toString()+"]";
-		} catch (SQLException e) {
-			log.warn("Exception determining databaseinfo", e);
-		}
-		return dsinfo;
+		throw new IllegalStateException("Datasource should always be of type TransactionalDbmsSupportAwareDataSourceProxy, found: " + getDatasource().getClass().getName());
 	}
 
 	public void setDbmsSupportFactory(IDbmsSupportFactory dbmsSupportFactory) {
 		this.dbmsSupportFactory=dbmsSupportFactory;
 	}
 
 	public IDbmsSupport getDbmsSupport() {
-		if (dbmsSupport==null) {
-			try {
+		if (dbmsSupport != null) {
+			return dbmsSupport;
+		}
+		try {
+			if (datasource instanceof TransactionalDbmsSupportAwareDataSourceProxy) {
+				Map<String, String> md = ((TransactionalDbmsSupportAwareDataSourceProxy) datasource).getMetaData();
+				dbmsSupport = dbmsSupportFactory.getDbmsSupport(md.get("product"), md.get("product-version"));
```

Comment:
I think you can get the instance directly from the DataSource in 7.9 and don't need to extract all this meta-data and make the other method public?

Reply:
Ok, will do that.